provisioning_templates/snippet/redhat_register.erb

```diff
@@ -100,6 +109,25 @@ snippet: true
     fi
   <% end %>
 
+  <% if host_param('syspurpose_role') %>
+      syspurpose set-role "<%= host_param('syspurpose_role') %>"
+  <% end %>
+
+  <% if host_param('syspurpose_usage') %>
+      syspurpose set-usage "<%= host_param('syspurpose_usage') %>"
+  <% end %>
+
+  <% if host_param('syspurpose_sla') %>
+      syspurpose set-sla "<%= host_param('syspurpose_sla') %>"
+  <% end %>
+
+  <% if host_param('syspurpose_add_ons') %>
+      <% add_ons = syspurpose_add_ons.split(',').map(:strip) %>
+      <% add_ons.each do |add_on| %>
+          syspurpose add-addons "<%= add_on %>"
```

Comment:
Slightly weird to me that `add-addons` only accepts a single value. It should either have been singular or accepted multiple values.

Reply:
Sorry I should've marked this WIP, I just got a hold of a build that has the features so I need to explore how it operates.  add-addons ostensibly takes multiple values, so I'll update that, as well as checking if the tool exists.